### ISSUE TITLE
Out of Memory Exception on Android

### ISSUE THREAD

**Issue by ggomaeng** [terrylinla/react-native-sketch-canvas#114]
When I wrap the SketchCanvas component with a <Animated.View> component and use dynamic width and height, I run out of memory on line 311 of SketchCanvas.java. Any pointers would be really appreciated.

**Comment by ShaMan123** [comment]
+1
I've been trying to solve this. Didn't notice it's to do with `Animated.View`, interesting.

Take a look at my fork and this [comment](https://stackoverflow.com/questions/16500379/view-too-large-to-fit-into-drawing-cache-when-calling-getdrawingcache). I've managed to reduce OOM to nil but now I'm receiving a warning and nothing renders.
```SketchCanvas not displayed because it is too large to fit into a software layer (or drawing cache), needs 9006080 bytes, only 3686400 available``` 

The problem is in `onSizeChanged` in native android code, where all the allocation is being done.
When using `Animated` `useNativeDriver=false` the view re-renders for each change, meaning, to my understanding, `onSizeChanged` gets called a lot in a very small amount of time resulting in OOM.

If I were you I would try a different solution that **uses the native driver** 
*OR* consider:
1. saving the sketch
2. displaying the output as an `Animated.Image`
3. running the animation 
4. rendering the sketch after the animation has finished.

**Comment by ggomaeng** [comment]
@ShaMan123 an123 thanks for the follow up! However, I was trying to implement an infinitely scrollable  canvas, and I ended up moving over to react-native-svg for performance reasons.

**Comment by ShaMan123** [comment]
and it's better? I tried react-native-svg and it was horrible

**Comment by ggomaeng** [comment]
For infinitely scrollable canvas, I found it to be more suitable since each path has an offset prop that I can link to the scroll offset of the scrollview. So the size of the canvas actually is always fixed, and when the scroll happens, the path elements are just shifted.

For implementing svg canvas, I used this [implementation](https://github.com/matiascaputti/react-native-paint).

**Comment by ShaMan123** [comment]
putting in a `ScrollView` didn't do the job?

**Comment by ggomaeng** [comment]
No, because the height of the canvas would have to expand in order to cover the offset of the scroll, which would put us back to this issue again.

**Comment by ShaMan123** [comment]
I think I understand.
I tried to render each path on a different canvas all rendered on top of each other behaving like layers and used `Animated` to do stuff.
I wonder if now my solution holds.

**Comment by ggomaeng** [comment]
That was my initial approach, but it introduces two problems:
1) Each canvas have different touch events, so drawing between canvases results in an awkward stop.
2) Memory issue on some devices, especially Android
